### PR TITLE
Make updated tracing scope/begin/end/instant use char* instead of enums

### PR DESCRIPTION
**Pull request by andy31415**
After discussing with users of tracing, it seems that there is a desire to make tracing easier to add at the expense of backend multiplexing flexibility.

Changes:
  - make tracing take label & group as const char *
  - Tracing is controlled independently from data logging: can have a separate config for tracing macros which would be in use for backends that want or need const char * strings (e.g. pw_trace or perfetto)

### CODE REVIEW

File: src/app/clusters/general-commissioning-server/general-commissioning-server.cpp [hunk 1]
```diff
@@ -156,7 +156,7 @@ bool emberAfGeneralCommissioningClusterArmFailSafeCallback(app::CommandHandler *
                                                            const Commands::ArmFailSafe::DecodableType & commandData)
 {
     MATTER_TRACE_EVENT_SCOPE("ArmFailSafe", "GeneralCommissioning");
-    MATTER_TRACE_SCOPE(::chip::Tracing::Scope::GeneralCommissioning_ArmFailSafe);
+    MATTER_TRACE_SCOPE("ArmFailSafe", "GeneralCommissioning");
```
Comment (mspang) [inline src/app/clusters/general-commissioning-server/general-commissioning-server.cpp:159]: Seems like it should be possible to transition these without duplicating. Thoughts?

Comment (andy31415) [reply src/app/clusters/general-commissioning-server/general-commissioning-server.cpp:159]: My plan was/is to fully replace it once I have functional equivalence. I found not users of pw_trace, but I did find users of perfetto, so I was planning to switch over when the perfetto backend is included and functional.

**Review by mspang — APPROVED** [review]

File: src/tracing/BUILD.gn [hunk 2]
```diff
@@ -21,6 +21,14 @@ config("tracing_enabled") {
   defines = [ "MATTER_TRACING_ENABLED" ]
 }
 
+config("tracing_multiplexed") {
+  defines = [ "MATTER_TRACE_MULTIPLEXED" ]
```
Comment (mspang) [inline src/tracing/BUILD.gn:25]: It is be preferable to generate a header with buildconfig_header to facilitate crossing build systems.

**Review by woody-apple — APPROVED** [review]

File: src/tracing/BUILD.gn (hunk 2 above)
Comment (andy31415) [reply src/tracing/BUILD.gn]: Done.

**Pull request merged**